core/src/main/java/org/infinispan/marshall/persistence/impl/PersistenceMarshallerImpl.java

```diff
@@ -82,7 +82,9 @@ public void start() {
          userMarshaller.start();
       }
 
-      register(new PersistenceContextInitializerImpl());
+      // TODO can we move this to the registry?
+      String messageName = PersistenceContextInitializer.getFqTypeName(MarshallableUserObject.class);
+      ctxRegistry.addMarshaller(MarshallerType.PERSISTENCE, new MarshallableUserObject.Marshaller(messageName, getUserMarshaller()));
```

Comment:
@anistor This is a hack, but to allow the `MarshallableUserObject` to be usable by the protostream-processor, I generate the proto schema/marshaller as normal but then override the marshaller with ^ `RawProtobufMarshaller` implenentation so that it can delegate to the configured user marshaller. The `SerializationContextRegistryImpl` ensures that the generated marshaller is always overridden by non-core modules.